src/NServiceBus.Transport.Msmq/MsmqMessageDispatcher.cs

```diff
@@ -77,6 +79,8 @@ void ExecuteTransportOperation(TransportTransaction transaction, UnicastTranspor
 
                         var label = GetLabel(message);
 
+                        await Task.Yield();
```

Comment:
@bording I didn't because there is some validation happening before the Yield. If you choose to do this then do this as early as possible? Is there some optimization by making it the first statement of the method?